File: Structural/Lab7/Typescript/src/program.ts
import { Customer } from './Entities/customer';
import { ApartmentBuilder } from './Builders/apartment_builder';
import { PriceAdapter } from './price_adapter';

class Program {
    static main(args: string[]): void {
        console.log("Start testing... \n\n");
        const customer = new Customer();
        const apartmentBuilder = new ApartmentBuilder();
        const apartment = apartmentBuilder.setBalcony().setParking().setAddress("XZ").getEstate();
        const adapter = new PriceAdapter(apartment);
        const price = 300000;
        console.log(`Sell apartment with price: ${price} UAH`);
        customer.setToBuy(adapter, price);
    }
}

Program.main([]);

File: Structural/Lab7/C#/Program.cs
using System;
using System.Diagnostics;
using System.Net.WebSockets;
using System.Reflection.Emit;
using System.Runtime.InteropServices;
using C_.Builders;
using C_.Entities;

namespace C_
{    
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Start testing... \n\n");
            Customer customer = new Customer();
            ApartmentBuilder apartmentBuilder = new ApartmentBuilder();
            var apartment = apartmentBuilder.SetBalcony().SetParking().SetAddress("XZ").GetEstate();
            var price = 300000;
            PriceAdapter adapter = new PriceAdapter(apartment);
            Console.WriteLine($"Sell apartment with price: {price} UAH");
            customer.SetToBuy(adapter, 300000);
        }

    }
}
